### PR TITLE
ME-77 API - add source type

### PR DESCRIPTION
**Pull request by arfenis**
- Create CRUD workflow for memo source types.

- Create migration for the table source_types

- Update schema file based in the actual database tables

### CODE REVIEW

File: api/controllers/SourceTypesController.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    /**
+     * set objects.
+     *
+     * @return void
+     */
+    public function onConstruct()
+    {
+        $this->model = new SourceTypes();
+        $this->softDelete = 1;
```
Comment (kaioken) [inline api/controllers/SourceTypesController.php]: @arfenis parent has this already specify, no?

File: storage/db/migrations/20200116153902_source_types.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                'row_format' => 'DYNAMIC',
+            ])
+            ->addColumn('id', 'integer', [
+                'null' => false,
+                'limit' => MysqlAdapter::INT_REGULAR,
+                'identity' => 'enable',
+            ])
+            ->addColumn('name', 'string', [
+                'null' => true,
+                'default' => null,
```
Comment (kaioken) [inline storage/db/migrations/20200116153902_source_types.php]: @arfenis this is a bug on the script that genrates, remove all taht say default => null

File: storage/db/migrations/20200116153902_source_types.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                'primary_key' => ['id'],
+                'engine' => 'InnoDB',
+                'encoding' => 'utf8mb4',
+                'collation' => 'utf8mb4_unicode_520_ci',
+                'comment' => '',
+                'row_format' => 'DYNAMIC',
+            ])
+            ->changeColumn('source_link', 'string', [
+                'null' => false,
+                'default' => '\'\'\'\'\'\'',
```
Comment (kaioken) [inline storage/db/migrations/20200116153902_source_types.php]: another bug on the app, update this and change default to ``

File: api/controllers/SourceTypesController.php (hunk 1 above)
Comment (arfenis) [reply api/controllers/SourceTypesController.php]: @kaioken  No, as far as i known.

Comment (kaioken) [reply api/controllers/SourceTypesController.php]: @arfenis yep https://github.com/bakaphp/canvas-core/blob/master/src/Api/Controllers/BaseController.php#L24 :D

Comment (kaioken) [reply api/controllers/SourceTypesController.php]: @arfenis yep https://github.com/bakaphp/canvas-core/blob/master/src/Api/Controllers/BaseController.php#L24 :D

Comment (arfenis) [reply api/controllers/SourceTypesController.php]: Ah ok, you were refereed to the softDelete not the onConstruct in general.

**Pull request merged**